src/Searchable/AggregatorObserver.php

```diff
@@ -85,12 +85,14 @@ public function saved($model): void
      */
     public function deleted($model): void
     {
-        if (static::syncingDisabledFor($model)) {
+        if (!$model->wasSearchableBeforeDelete()) {
```

Comment:
This check here won't always work. Aggregators can contain classes that don't have the `Searchable` trait on them. Without this trait, this check will fail because the method is not defined. We should probably add a check on whether or not the model includes the `Searchable` trait (you can see how [here](https://www.algolia.com/doc/framework-integration/laravel/advanced-use-cases/multiple-models-in-one-index/?client=php#conditionally-sync-an-aggregator)).

Reply:
I didn't know that. Are you sure aggregators work if the underlying models don't have the Searchable trait? The docs for scout extended recommend disabling syncing for models if you don't want them to create their own individual model indexes. If what you say is correct, that wouldn't be needed.

Reply:
Yes, this is something that's possible to do. This is not well documented, but the tests ensure this works as well (in `AggregatorTest.php`, we use the `News` aggregator that includes the `Post` model, which doesn't include the `Searchable` trait).

Reply:
Interesting. In that case, maybe best to check if the method exists rather than whether the Searchable trait is being used. That way an aggregated class without the Searchable trait can still implement a wasSearchableBeforeDelete method if wanted.